CompareRaftMembers/DiffStream.js

```diff
@@ -0,0 +1,173 @@
+const stream = require('stream');
+
+const BucketStream = require('./BucketStream');
+
+/**
+ * Output differences between the input stream, consisting of
+ * { key: string, value: string } objects sorted by key, and a bucketd
+ * interface that can be queried via HTTP for ranges of keys from
+ * given buckets.
+ *
+ * Input items are meant to be coming from a raw leveldb database
+ * listing, transformed via a DBListStream object to be comparable with
+ * bucketd listing.
+ *
+ * @class DiffStream
+ */
+class DiffStream extends stream.Transform {
+    /**
+     * @constructor
+     * @param {object} params - constructor params
+     * @param {string} params.bucketdHost - bucketd host name or IP address
+     * @param {number} params.bucketdPort - bucketd API port
+     * @param {number} [params.maxBufferSize=1000] - maximum number of
+     * items to bufferize
+     * @param {class} [params.BucketStreamClass=BucketStream] -
```

Comment:
why is this needed in the params instead of modifying BucketStreamClass after instantiation in the tests?

Reply:
It's because the instanciation happens internally, potentially multiple times, from the `_compareInputBufferWithBucketd` helper.